xtext-gradle-plugin/src/main/java/org/xtext/gradle/tasks/XtextExtension.xtend

```diff
@@ -129,10 +133,10 @@ class Language implements Named {
 
 @Accessors
 class GeneratorConfig {
-	boolean suppressWarningsAnnotation = true
-	String javaSourceLevel
-	val GeneratedAnnotationOptions generatedAnnotation = new GeneratedAnnotationOptions
-	val NamedDomainObjectContainer<Outlet> outlets
+	@Input boolean suppressWarningsAnnotation = true
+	@Input String javaSourceLevel
+	@Nested val GeneratedAnnotationOptions generatedAnnotation = new GeneratedAnnotationOptions
```

Comment:
why no `@Optional` needed?

Reply:
Because this field is final, thus always set. Or did you mean the line above (javaSourceLevel)?

Reply:
no i meant generatedAnnotation